### ISSUE TITLE
Segfault when using incorrect(?) alias syntax

### ISSUE THREAD

**Issue by falsifian** [neomutt/neomutt#2677]
I used the following syntax to define an alias in my `.neomuttrc` file:

    alias foo "Foo <foo@bar.baz>"

When I press "m" to send a mail, and respond to the `To:` prompt with "foo", neomutt dies with a segmentation fault.

If I remove the double quotes, the alias works.

## Expected Behaviour

I expect neomutt to either send the mail to `Foo <foo@bar.baz>` or tell me there's a syntax error in my `.neomuttrc`.

(Quickly scanning the man page, I see "` Single quotes ("'") and double quotes (""") can be used`...", so I would expect it to just work.)

## Actual Behaviour

neomutt exits with a segmentation fault.

If NeoMutt crashed, did your OS create a 'coredump' file?

Yes.

## Steps to Reproduce

Here's a way I can reproduce with a minimal configuration.

Create a file named `config` with only the following:

    alias foo "Foo <foo@bar.baz>"

Run `neomutt -F config`. I answer "no" to the "/home/$USER/Mail does not exist" prompt. 

Type "m". At the `To:` prompt, type `foo`. Result:

    cur$ neomutt -F config
    Segmentation fault (core dumped)

## How often does this happen?

- Always

## When did it start to happen?

The first time I added an `alias` line with double quotes to my `.neomuttrc`.

## NeoMutt Version

```
NeoMutt 20200821
Copyright (C) 1996-2020 Michael R. Elkins and others.
NeoMutt comes with ABSOLUTELY NO WARRANTY; for details type 'neomutt -vv'.
NeoMutt is free software, and you are welcome to redistribute it
under certain conditions; type 'neomutt -vv' for details.

System: OpenBSD 6.8 (amd64)
ncurses: ncurses 5.7.20081102 (compiled with 5.7.20081102)
libiconv: 1.16
libidn2: 2.3.0 (compiled with 2.3.0)
GPGME: 1.14.0
LibreSSL: LibreSSL 3.2.2
OpenSSL: LibreSSL 3.2.2
storage: kyotocabinet, tdb

Configure options: --mandir=/usr/local/man --with-ui=ncurses --docdir=/usr/local/share/doc/neomutt --with-lock=flock --idn=0 --idn2 --fmemopen --ssl --tdb --debug --autocrypt --gpgme --sqlite --sasl --kyotocabinet

Compilation CFLAGS: -O2 -pipe -g  -I/usr/local/include -std=c99 -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -DNCURSES_WIDECHAR

Default options:
  +attach_headers_color +compose_to_sender +compress +cond_date +debug 
  +encrypt_to_self +forgotten_attachments +forwref +ifdef +imap +index_color 
  +initials +limit_current_thread +multiple_fcc +nested_if +new_mail +nntp +pop 
  +progress +quasi_delete +regcomp +reply_with_xorig +sensible_browser +sidebar 
  +skip_quoted +smtp +status_color +timeout +tls_sni +trash 

Compile options:
  +autocrypt +bkgdset +color +curs_set -fcntl +flock +fmemopen +futimens 
  +getaddrinfo -gnutls +gpgme -gss +hcache -homespool +idn -inotify 
  -locales_hack -lua +meta -mixmaster +nls -notmuch +openssl +pgp +regex +sasl 
  +smime +sqlite +start_color +sun_attachment +typeahead 
MAILPATH="/var/mail"
PKGDATADIR="/usr/local/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/usr/local/etc"

To learn more about NeoMutt, visit: https://neomutt.org
If you find a bug in NeoMutt, please raise an issue at:
    https://github.com/neomutt/neomutt/issues
or send an email to: <neomutt-devel@neomutt.org>
```

Also happens with `20200320` (which is the version in the OpenBSD 6.7 stable branch).

## Extra Info

* Operating System and its version

OpenBSD current (just upgraded today). It also happens on 6.7-stable.

* Were you using multiple copies of NeoMutt at once?

No.

* Were you using 'screen' or 'tmux'?

Yes. (tmux)

* Is your email local (maildir) or remote (IMAP)?

It happens either way. My normal configuration uses IMAP, and it happened with that. The minimal one-line config above (presumably) doesn't use IMAP.

**Comment by jindraj** [comment]
When trying to reproduce I'm getting following warning and alias isn't created. But that's probably not your case, since your neomutt crashes when the alias is used.

I think the issue are the quotes around the whole NAME + ADDRESS
`Warning: Bad address '"Foo <foo@bar.baz>"' in alias 'foo'`

Try just following:
```
alias foo "Foo" <foo@bar.baz>
```

**Comment by gahr** [comment]
This was fixed by #2638, which is in 20200925.
